### ISSUE TITLE
ActiveRecord Enum broken in Rails 6.0.4

### ISSUE THREAD

**Issue by kevin-smartpatients** [rails/rails#42527]
I just upgraded rails from 6.0.3.7 to  6.0.4 and now one (just one) of my enums fails during initializing with:

> Minitest::UnexpectedError: ArgumentError: wrong number of arguments (given 0, expected 1..3) 

which occurs on this line in ActiveRecord::Enum:

```
      def detect_negative_enum_conditions!(method_names)
        return unless logger
```

I believe it is caused by this change https://github.com/rails/rails/pull/36380

I assume the `logger` method is overridden somewhere in another module but I can't figure out where.  

My enum is defined in a concern. It only fails on this enum. All other enums work fine.

```
module Links::Previews
  extend ActiveSupport::Concern

  included do
    enum fetch_status: {
        ready_to_fetch: 0,
        fetching: 1,
        fetched: 2,
        failed: 3,
    }
  end
end
```

### Steps to reproduce
I'm not able to reproduce this in a standalone project.

### System configuration
**Rails version**: 6.0.4
**Ruby version**: 2.6.6

**Comment by rafaelfranca** [comment]
Rails doesn't override `logger` anywhere, so it doesn't seems to be an issue in Rails. Can you check if in the models that include that concern you don't have something that override `logger`?

**Comment by kevin-smartpatients** [comment]
> Can you check if in the models that include that concern you don't have something that override logger?

There is nothing that overrides `logger`. It's quite a simple class. There is not a lot going on in there.

This is failing in a test. I'll try it in the actual app to see if just a test problem.

**Comment by casperisfine** [comment]
> I assume the `logger` method is overridden somewhere in another module but I can't figure out where.

If you are able to reproduce this locally:

  - `bundle open activerecord`
  - before `return unless logger` insert `p method(:logger)`
  - make sure spring is restarted

Next time you run your test it should print the `Method#inspect` which includes the method location. If something override `logger` like you suspect that will tell you where it happens.

**Comment by kevin-smartpatients** [comment]
> Next time you run your test it should print the `Method#inspect` which includes the method location. If something override `logger` like you suspect that will tell you where it happens.

Excellent, thank you! I found the problem. 

`HTTParty` (which I am including) is overriding logger. I can fix it by calling the method I want directly instead of including it. 

Now that I think of it, I saw this error when I tried to upgrade to Rails 6.1 several months ago. I did not have the bandwidth to investigate at the time so I stuck with 6.0. Now I know the problem, I can upgrade.

Thanks all for your prompt and excellent help!
